src/language-js/needs-parens.js

```diff
@@ -531,7 +531,13 @@ function needsParens(path, options) {
         (grandParent.init === parent || grandParent.update === parent)
       ) {
         return false;
-      } else if (name === "value" && parent.type === "Property") {
+      } else if (
+        name === "value" &&
+        parent.type === "Property" &&
+        grandParent &&
+        grandParent.type === "ObjectPattern" &&
+        grandParent.properties.includes(parent)
```

Comment:
BTW, `ObjectPattern` doesn't have any other child nodes, only `properties`, so this `includes` seems unneeded.

Reply:
I know that, just for safety.

Reply:
By the way, do you think adopt `esquery` is a good idea?

So we can do something like

```js
isNodeMatch(node, 'ObjectPattern > Property.properties > AssignmentExpression.value');
```
I'm a little worry about performance

Reply:
Looks interesting, but I have doubts about the performance too.

Reply:
Another off topic one, I have been thinking about use esquery selector in `embeddedLanguageFormatting` option few weeks ago, :smile: